spring-pulsar/src/main/java/org/springframework/pulsar/listener/DefaultPulsarMessageListenerContainer.java

```diff
@@ -541,12 +546,29 @@ public void acknowledge() {
 
 		@Override
 		public void acknowledge(MessageId messageId) {
-			throw new UnsupportedOperationException();
+			try {
```

Comment:
Wdyt about 

1. moving this out to a method that gets used by here and the list version below.
2. log the exception message and state that the ack failed and that we are now instead nacking? If so, what level should that be at DEBUG/INFO/WARN? Will this happen often or is it rare?

Reply:
https://github.com/spring-projects-experimental/spring-pulsar/issues/120 turns out to be a bigger fish to fry than this section in this PR. We can handle this post M1.